corehq/apps/cloudcare/views.py

```diff
@@ -131,7 +131,7 @@ def get_web_apps_available_to_user(self, domain, user):
             # User has access via domain mirroring
             pass
         if role:
-            apps = [_format_app(app) for app in apps if role.permissions.view_web_app(app)]
+            apps = [_format_app(app) for app in apps if role.permissions.view_web_app(app.master_id)]
```

Comment:
This is causing 500s.

```
  File "/home/cchq/www/icds-staging/releases/2020-07-17_07.57/corehq/apps/accounting/decorators.py", line 124, in wrapped
    return fn(request, *args, **kwargs)
  File "/home/cchq/www/icds-staging/releases/2020-07-17_07.57/python_env-3.6/lib/python3.6/site-packages/django/utils/decorators.py", line 63, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/home/cchq/www/icds-staging/releases/2020-07-17_07.57/corehq/apps/cloudcare/views.py", line 106, in dispatch
    return super(FormplayerMain, self).dispatch(request, *args, **kwargs)
  File "/home/cchq/www/icds-staging/releases/2020-07-17_07.57/python_env-3.6/lib/python3.6/site-packages/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/cchq/www/icds-staging/releases/2020-07-17_07.57/corehq/apps/cloudcare/views.py", line 186, in get
    return self.get_main(request, domain)
  File "/home/cchq/www/icds-staging/releases/2020-07-17_07.57/corehq/apps/cloudcare/views.py", line 195, in get_main
    apps = self.get_web_apps_available_to_user(domain, restore_as)
  File "/home/cchq/www/icds-staging/releases/2020-07-17_07.57/corehq/apps/cloudcare/views.py", line 134, in get_web_apps_available_to_user
    apps = [_format_app(app) for app in apps if role.permissions.view_web_app(app.master_id)]
  File "/home/cchq/www/icds-staging/releases/2020-07-17_07.57/corehq/apps/cloudcare/views.py", line 134, in <listcomp>
    apps = [_format_app(app) for app in apps if role.permissions.view_web_app(app.master_id)]
AttributeError: 'dict' object has no attribute 'master_id'
```

Reply:
Hey @snopoke, 
Do you think if I add a check if `app.master_id` is present then use `app.master_id` else user `app_id` can be a possible fix to it?

Reply:
Simon is offline today.  It looks like the problem is that `app` is an unwrapped JSON app document.  Since you're only dealing with primary applications anyways, you can probably use just `app['_id']`

Reply:
The apps are coming from `fetch_app` which returns a build, so I think it should be `app[copy_of]`. Or, to be cautious, this could use the same calculation as `master_id`, that is, `app['copy_of'] or app['_id']`

Reply:
Oh yes you're right, my bad

Reply:
Thanks @esoergel  and @orangejenny for your suggestions. 
I have updated the code with `app['copy_of'] or app['_id']`.